code/modules/reagents/Chemistry-Recipes.dm

```diff
@@ -606,13 +606,68 @@
 	name = "Solid Plasma"
 	id = "solidplasma"
 	result = null
-	required_reagents = list(IRON = 5, FROSTOIL = 5, PLASMA = 20)
+	required_reagents = list(SILICATE = 10, FROSTOIL = 10, PLASMA = 20)
 	result_amount = 1
 
 /datum/chemical_reaction/plasmasolidification/on_reaction(var/datum/reagents/holder, var/created_volume)
 	var/location = get_turf(holder.my_atom)
 	new /obj/item/stack/sheet/mineral/plasma(location)
 
+/datum/chemical_reaction/ironsolidification
+	name = "Solid Metal"
+	id = "solidmetal"
+	result = null
+	required_reagents = list(SILICATE = 10, FROSTOIL = 10, IRON = 20)
+	result_amount = 1
+
+/datum/chemical_reaction/ironsolidification/on_reaction(var/datum/reagents/holder, var/created_volume)
+	var/location = get_turf(holder.my_atom)
+	new /obj/item/stack/sheet/metal(location)
+
+/datum/chemical_reaction/silversolidification
+	name = "Solid Silver"
+	id = "solidsilver"
+	result = null
+	required_reagents = list(SILICATE = 10, FROSTOIL = 10, SILVER = 20)
+	result_amount = 1
+
+/datum/chemical_reaction/silversolidification/on_reaction(var/datum/reagents/holder, var/created_volume)
+	var/location = get_turf(holder.my_atom)
```

Comment:
Instead of making a lot of copypaste code, you should instead make 

/datum/chemical_reaction/solidification/on_reaction()

and then create children of solidification, which inherit its proc, you can use result to store the path of the sheet you want to create

For an example, look at vaporization code